### PR TITLE
Document flutter::DartServiceIsolate.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by chinmaygarde** [comment]
Minor doc only change to land on red to unblock LUCI that is stuck on an infra flake.

File: runtime/dart_service_isolate.h [hunk 1]
```diff
@@ -11,16 +11,54 @@
...
 namespace flutter {
 
+//------------------------------------------------------------------------------
+/// @brief      Utility methods for interacting with the DartVM managed service
+///             isolate present in debug and profile runtime modes.
+///
 class DartServiceIsolate {
  public:
+  //----------------------------------------------------------------------------
+  /// The handle used to refer to callbacks registered with service isolate.
```
Comment (jason-simmons) [inline runtime/dart_service_isolate.h]: typo: "with the service isolate"

File: runtime/dart_service_isolate.h [hunk 2]
```diff
@@ -29,14 +67,32 @@ class DartServiceIsolate {
...
+  //----------------------------------------------------------------------------
+  /// @brief      Add a callback that will get invoked when the observatory
+  ///             starts up. If the observatory has already started before this
+  ///             call is made, the callback is invoked immediately.
+  ///
+  ///             This method is thread safe.
+  ///
+  /// @param[in]  callback  The callback with information about the observatory.
+  ///
+  /// @return     A handle for the callback that can be used layer in
```
Comment (jason-simmons) [inline runtime/dart_service_isolate.h]: typo: "later"

File: runtime/dart_service_isolate.h (hunk 1 above)
Comment (chinmaygarde) [reply runtime/dart_service_isolate.h]: Done.

File: runtime/dart_service_isolate.h (hunk 2 above)
Comment (chinmaygarde) [reply runtime/dart_service_isolate.h]: Done.

**Pull request merged**